Enhance Python linting settings and add non-Steam game support

peter9811 opened the pull request:
Improve VSCode settings for Python linting and type checking. Introduce functionality to include non-Steam games and create shortcuts for them.

peter9811 committed:
chore: enhance VSCode settings for Python linting and type checking

peter9811 committed:
feat: add functionality to include non-Steam games and create shortcuts

peter9811 committed:
Merge remote-tracking branch 'upstream/master'

peter9811 committed:
feat: enhance non-Steam game input with filename sanitization and optional arguments

JeeZeh commented:
Thanks, but I'm not sure this addition really makes sense for the tool. It's much quicker to create shortcuts for non-Steam games directly from those applications, no?

peter9811 commented:
Yes. But following the request of one of the "issues"
Is better have it and don't use, and need it and don't have it, haha

Give it a test, if you're not happy, is okay

peter9811 committed:
feat: add platform-specific shortcut creation for Windows and Linux

peter9811 committed:
fix: update LICENSE formatting, improve README image syntax, and enhance error handling in steam_shortcuts.py

peter9811 commented:
@JeeZeh better now?

dependabot[bot] committed:
chore(deps): bump urllib3 from 1.26.15 to 1.26.19

Bumps [urllib3](https://github.com/urllib3/urllib3) from 1.26.15 to 1.26.19.
- [Release notes](https://github.com/urllib3/urllib3/releases)
- [Changelog](https://github.com/urllib3/urllib3/blob/main/CHANGES.rst)
- [Commits](https://github.com/urllib3/urllib3/compare/1.26.15...1.26.19)

---
updated-dependencies:
- dependency-name: urllib3
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump pillow from 10.0.0 to 10.3.0

Bumps [pillow](https://github.com/python-pillow/Pillow) from 10.0.0 to 10.3.0.
- [Release notes](https://github.com/python-pillow/Pillow/releases)
- [Changelog](https://github.com/python-pillow/Pillow/blob/main/CHANGES.rst)
- [Commits](https://github.com/python-pillow/Pillow/compare/10.0.0...10.3.0)

---
updated-dependencies:
- dependency-name: pillow
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

peter9811 committed:
Merge pull request #2 from peter9811/dependabot/pip/pillow-10.3.0

chore(deps): bump pillow from 10.0.0 to 10.3.0

peter9811 committed:
Merge pull request #1 from peter9811/dependabot/pip/urllib3-1.26.19

chore(deps): bump urllib3 from 1.26.15 to 1.26.19

peter9811 committed:
fix(steam_shortcuts): improve error handling and code robustness

- Add numeric validation for Steam ID to prevent invalid API requests
- Enhance error handling for Steam API responses with detailed error messages
- Fix Windows registry path escaping and improve path handling
- Add missing fields in game dictionary and improve type safety
- Improve shortcut creation with better file naming and error logging

dependabot[bot] committed:
chore(deps): update pillow requirement

Updates the requirements on [pillow](https://github.com/python-pillow/Pillow) to permit the latest version.

Updates `pillow` to 11.2.1
- [Release notes](https://github.com/python-pillow/Pillow/releases)
- [Changelog](https://github.com/python-pillow/Pillow/blob/main/CHANGES.rst)
- [Commits](https://github.com/python-pillow/Pillow/compare/10.3.0...11.2.1)

---
updated-dependencies:
- dependency-name: pillow
  dependency-version: 11.2.1
  dependency-type: direct:production
  dependency-group: pip
...

Signed-off-by: dependabot[bot] <support@github.com>

peter9811 committed:
Merge pull request #3 from peter9811/dependabot/pip/pip-a5d3283145

chore(deps): update pillow requirement from <11.0.0,>=10.3.0 to >=10.3.0,<12.0.0 in the pip group across 1 directory

peter9811 committed:
chore: simplify dependencies in requirements.txt

Remove version constraints for Pillow based on Python versions and simplify version specifications for all dependencies to improve maintainability and reduce complexity.

peter9811 commented:
@JeeZeh working like charm, now, haha

JeeZeh commented:
> Is better have it and don't use, and need it and don't have it

Hey, sorry for not getting back to this. I disagree on this part, I don't think we need support for manually creating non-steam shortcuts as its out of the scope of this tool.

Regarding the Linux support, have you tested this on a Linux machine? I'm wary that not all distros will expect shortcuts in the same location, maybe this should be optionally overridable. I can address that separately, but I'd like to confirm that what you are comitting as worked on at least one popular distro.

Thanks.

peter9811 committed:
refactor: remove unused imports and comment out non-Steam games code

This commit removes unused imports such as `Optional` and `path`, and comments out the non-Steam games functionality to clean up the codebase. The changes aim to improve code maintainability and readability without altering the program's behavior.

peter9811 committed:
refactor(steam_shortcuts): improve folder path handling with XDG compliance

The changes introduce XDG-compliant path handling for the shortcuts folder and add an environment variable override for custom directory paths. This improves compatibility with XDG standards and allows users to customize the shortcuts directory via the STEAM_SHORTCUTS_DIR environment variable.

peter9811 commented:
> Regarding the Linux support, have you tested this on a Linux machine? I'm wary that not all distros will expect shortcuts in the same location, maybe this should be optionally overridable. I can address that separately, but I'd like to confirm that what you are comitting as worked on at least one popular distro.

I commented the non-Steam, because you have the reason, maybe is too slow, haha
About Linux, I tried on WSL on Windows, worked for me, I'm not very familiar with Linux, haha

JeeZeh commented:
Closing this as I don't think these changes quite fit the purpose of the project. Thanks for your interest!
